macos/FluentUI/Button/Button.swift

```diff
@@ -247,6 +279,30 @@ open class Button: NSButton {
 			layer.backgroundColor = backgroundColorRest?.cgColor
 			layer.borderColor = borderColorRest?.cgColor
 		}
+
+		if usesBorderShadows {
+			// Although we're using an AssetCatalog color for the shadows, which has a dynamic dark/light
+			// appearance specified, we have to manually update the shadowColor property of the layer
+			// since it's a CGColor and not an NSColor, and CGColorRef objects don’t adapt to environment
+			// changes
+			updateShadowLayer(shadowLayer: firstOuterDropShadowLayer,
+							  shadowColor: ButtonColor.firstOuterDropShadow?.cgColor)
+			updateShadowLayer(shadowLayer: secondOuterDropShadowLayer,
+							  shadowColor: ButtonColor.secondOuterDropShadow?.cgColor)
+			updateShadowLayer(shadowLayer: innerShadowLayer,
+							  shadowColor: ButtonColor.innerShadow?.cgColor)
+		}
+	}
+
+	func updateShadowLayer(shadowLayer: CALayer?, shadowColor: CGColor?) {
```

Comment:
In this particular func, don't we always want shadowLayer and shadowColor to be non-optional? Can use if/lets in the caller to enforce that

Reply:
defeats the purpose of using usesBorderShadows, which was added to keep code cleaner, but it sounds like it sacrifices correctness?

Reply:
I think it's just another axis of keeping code cleaner - from what I've seen, persisting optional-ness longer in swift codepaths has abnormally large impact on code cleanliness given all the unwraps/"if let"s it forces